source/kubernetes/versions.rst

```diff
@@ -0,0 +1,116 @@
+
+For example:
+
+.. code-block:: bash
+
+  [major].[minor].[patch]
+  
+  v1.18.2
+  v1.17.5
+  v1.16.9
+
+Catalyst Cloud Kubernetes Service uses cluster template to manage each Kubernetes
+version and the matrix of addons running on top of Kubernetes cluster. And users
+should be able to see the Kubernetes version from the cluster template name. For
+example: *kubernetes-v1.16.9-prod-20200602*
+
+
+For more information, see `Kubernetes Release Versioning`_.
+
+.. _`Semantic Versioning`: http://semver.org/
+.. _`Kubernetes Release Versioning`: https://github.com/kubernetes/community/blob/master/contributors/design-proposals/release/versioning.md#kubernetes-release-versioning
+
+Kubernetes Versions Support Policy
+==================================
+
+Catalyst Cloud Kubernetes Service supports at least ``3`` minors versions. As long
+as there is a new minor version released, Catalyst Cloud Kubernetes Service will
+try to get it certified (pass the CNCF conformance test) and released in ``30``
+days. And then deprecate the oldest minor version. If the current 3 minor versions 
+are v1.17.x, v1.16.x and v1.15.x. Then as long as the new v1.18.x released, the
+v1.15.x versions(both the two patch versions) will be removed and out of supported.
+Here, out of support means whenever users ask for support, you will be asked
+to upgrade your clusters to supported versions first.
+
+Catalyst Cloud Kubernetes Service supports the latest ``2`` stable patch versions
+for each version. As long as there is a patch version released, the oldest patch
+version will be removed and out of supported. For example, if current versions
+supported for v1.16.x are v1.16.9 and v1.16.8, then v1.16.8 will be removed in
+favor of the release of v1.16.10.
+
+.. code-block:: bash
+
+    v1.18.2, v1.18.1, v1.17.5, v1.17.4, v1.16.9, v1.16.8
+
+Users should always aim to run the latest patch version for each minor version
+to get the latest security enhancements. For example, if the current Kubernetes
+cluster is running on v1.16.9 and the new patch version is v1.16.10, then it
+is highly recommended to upgrade to v1.16.10 as soon as possible.
+
+When there is a new patch version released, users have ``30`` days to upgrade
+to a supported patch version if cluster's current version is deprecated by the
+new version released.
+
+
+.. note::
+
+    If the cluster is running on a version which has been deprecated, then the
+    cluster is out of support.
+
+.. note::
+
+    Catalyst Cloud reserves the right to add/remove a new/existing cluster
+    template if there is a cirtical issue identified in the version without
+    further notice.
+
+****************************
+Upgrading Kubernetes version
+****************************
+
+When doing Kubernetes version upgrade, minor vesion cannot be skipped. For
+example, if the current cluster version is v1.16.x, then it's not allowed
+to upgrade to v1.18.x. You have to upgrade to v1.17.x and then do another
+upgrade to v1.18.x.
```

Comment:
Will this be enforced by the service? Ideally users will be prevented from skipping a version if the alternative is to end up with a cluster that is both broken due to a failed upgrade and unsupported because they didn't follow instructions.

Reply:
@evhan  We don't enforce it now. The upstream team is working on a feature to support this soon. We would like to ask users to follow this before we have a mechanism to enforce it, though technically skip version upgrade should work from the Magnum perspective. The main concern is that the compatibility of k8s api may break between minor versions.

Reply:
OK, that makes sense. Good to know that it's planned, but let's not wait!

Reply:
We're not waiting. I just need time to get it done :)